packages/sdk/src/catalog/CatalogClient.ts

```diff
@@ -1 +1,371 @@
+      });
+
+      const data = await this._client.post<{ file: File }>(queryString, {
+        body: JSON.stringify(payload),
+      });
+
+      return Promise.resolve(data.file);
+    } catch (error) {
+      return Promise.reject(error);
+    }
+  }
+
+  async listCatalogFiles(
+    props: ListCatalogFilesRequest & { enablePagination: true },
+  ): Promise<ListCatalogFilesResponse>;
+  async listCatalogFiles(
+    props: ListCatalogFilesRequest & { enablePagination: false },
+  ): Promise<File[]>;
+  async listCatalogFiles(
+    props: ListCatalogFilesRequest & { enablePagination: boolean },
+  ): Promise<ListCatalogFilesResponse | File[]> {
+    const { namespaceId, catalogId, pageSize, pageToken, enablePagination } =
+      props;
+
+    try {
+      const files: File[] = [];
+
+      const queryString = getQueryString({
+        baseURL: `/namespaces/${namespaceId}/catalogs/${catalogId}/files`,
+        pageSize,
+        pageToken,
+      });
+
+      const data =
+        await this._client.get<ListCatalogFilesResponse>(queryString);
+
+      if (enablePagination) {
+        return Promise.resolve(data);
+      }
+
+      files.push(...data.files);
+
+      if (data.nextPageToken) {
+        files.push(
+          ...(await this.listCatalogFiles({
+            namespaceId,
+            catalogId,
+            pageSize,
+            pageToken: data.nextPageToken,
+            enablePagination: false,
+          })),
+        );
+      }
+
+      return Promise.resolve(files);
+    } catch (error) {
+      return Promise.reject(error);
+    }
+  }
+
+  async deleteCatalogFile(props: DeleteCatalogFileRequest): Promise<void> {
+    const { fileUid } = props;
+
+    try {
+      await this._client.delete(`/catalogs/files?fileUid=${fileUid}`);
+      return Promise.resolve();
+    } catch (error) {
+      return Promise.reject(error);
+    }
+  }
+
+  async getFileDetails(
+    props: GetFileDetailsRequest,
+  ): Promise<GetFileDetailsResponse> {
+    const { namespaceId, catalogId, fileId } = props;
+
+    try {
+      const queryString = getQueryString({
+        baseURL: `/namespaces/${namespaceId}/catalogs/${catalogId}/files/${fileId}`,
+      });
+
+      const data = await this._client.get<{ file: File }>(queryString);
+      return Promise.resolve(data.file);
+    } catch (error) {
+      return Promise.reject(error);
+    }
+  }
+
+  async getFileContent(
+    props: GetFileContentRequest,
+  ): Promise<GetFileContentResponse> {
+    const { namespaceId, catalogId, fileUid } = props;
+
+    try {
+      const queryString = getQueryString({
+        baseURL: `/namespaces/${namespaceId}/catalogs/${catalogId}/files/${fileUid}/source`,
+      });
+
+      const data = await this._client.get<{ sourceFile: FileContent }>(
+        queryString,
+      );
+      return Promise.resolve(data.sourceFile);
+    } catch (error) {
+      return Promise.reject(error);
+    }
+  }
```

Comment:
@thewbuk This should follow the instill.tech API docs name which is `getCatalogSingleSourceOfTruthFile`

https://openapi.instill.tech/reference/artifactpublicservice_getsourcefile